source/agora/common/Deserializer.d

```diff
@@ -103,7 +103,7 @@ public void deserializePart (T)(ref T record, scope DeserializeDg dg)
 {
     import std.traits;
     OriginalType!T orig_val;
-    deserializePart(orig_val, dg);
+    deserializePart(cast(OriginalType!T) orig_val, dg);
```

Comment:
`orig_val` is already typed as `OriginalType!T`.

Reply:
Error: none of the overloads of deserializePart are callable using argument types (int, ubyte[] delegate(ulong size) @safe), candidates are:

Reply:
Don't use `OriginalType. Pass the member directly.

Reply:
Explained IRL.

So the actual issue was that `NetworkState` was another enum but it defaulted to `int`. So I suggested to make it `NetworkState : ubyte` in a separate commit, and getting rid of this cast here.